cave/utils/csv2rh.py

```diff
@@ -0,0 +1,250 @@
+        """
+        self.logger = logging.getLogger('cave.utils.csv2rh')
+        self.input_reader = InputReader()
+        self.pcs = pcs
+        self.train_inst = input_reader.read_instance_file(train_inst) if type(train_inst) == str else train_inst
+        self.test_inst =  input_reader.read_instance_file(test_inst) if type(test_inst) == str else test_inst
+        self.instance_features = input_reader.read_instance_features_file(instance_features) if type(instance_features) == str else instance_features
+        self.id_to_config = configurations if configurations else {}
+        self.cs = None
+
+        # Read in data
+        if isinstance(data, str):
+            csv_path = data
+            with open(csv_path, 'r') as csv_file:
+                csv_data = list(csv.reader(csv_file, delimiter=',',
+                                           skipinitialspace=True))
+            header, csv_data = csv_data[0], np.array([csv_data[1:]])[0]
+            self.data = pd.DataFrame(csv_data, columns=header)
+            self.data = self.data.apply(pd.to_numeric, errors='ignore')
+            self.logger.debug("Headers: " + str(list(self.data.columns.values)))
+        else:
+            self.data = data
+
+        # Expecting header as described in docstring
+        valid_values = ['seed', 'cost', 'time', 'status', 'config_id', 'instance_id']
+        if not len(self.data.columns) == len(set(self.data.columns)):
+            raise ValueError("Detected a duplicate in the columns of the "
+                             "csv-file \"%s\"." % csv_path)
+        for column_name in self.data.columns:
+            if not (column_name in valid_values or
+                    column_name.startswith('p_') or  # parameter
+                    column_name.startswith('i_')):  # instance feature
+                raise ValueError("%s not a legal column name in %s." % column_name, csv_path)
+
+        self._complete_configs()
+        self._complete_instances()
+        self.logger.debug("Found: seed=%s, cost=%s, time=%s, status=%s",
+                          'seed' in self.data.columns, 'cost' in self.data.columns,
+                          'time' in self.data.columns, 'status' in self.data.columns)
+
+        # Create RunHistory
+        rh = RunHistory(average_cost)
+        def add_to_rh(row):
+            new_status = self._interpret_status(row['status']) if 'status' in row else StatusType.SUCCESS
+            rh.add(config=self.id_to_config[row['config_id']],
+                   cost=row['cost'],
+                   time=row['time'] if 'time' in row else -1,
+                   status=new_status,
+                   instance_id=row['instance_id'] if 'instance_id' in row else None,
+                   seed=row['seed'] if 'seed' in row else None,
+                   additional_info=None,
+                   origin=DataOrigin.INTERNAL)
+
+        self.data.apply(add_to_rh, axis=1)
+        return rh
+
+    def get_cs(self):
```

Comment:
This method should be part of Pyimp. This would allow users to very easily use pyimp as stand-alone if they are only interested in parameter importance (and maybe don't have instance features).